src/ess/isissans/general.py

```diff
@@ -52,45 +75,68 @@ def default_parameters() -> dict:
-) -> RawDetector[RunType]:
+def to_detector_position_offset(
+    global_offset: DetectorBankOffset, beam_center: BeamCenter
+) -> DetectorPositionOffset[RunType]:
+    return DetectorPositionOffset[RunType](global_offset - beam_center)
+
+
+def to_monitor_position_offset(
+    global_offset: MonitorOffset[MonitorType],
+) -> MonitorPositionOffset[RunType, MonitorType]:
+    return MonitorPositionOffset[RunType, MonitorType](global_offset)
+
+
+def get_source_position(dg: LoadedFileContents[RunType]) -> SourcePosition[RunType]:
+    """Get source position from raw data."""
+    return SourcePosition[RunType](dg['data'].coords['source_position'])
+
+
+def get_sample_position(
+    dg: LoadedFileContents[RunType], offset: SampleOffset
+) -> SamplePosition[RunType]:
+    """Get sample position from raw data and apply user offset."""
+    return SamplePosition[RunType](
+        dg['data'].coords['sample_position'] + offset.to(unit='m')
+    )
+
+
+def get_detector_data(dg: LoadedFileContents[RunType]) -> NeXusDetector[RunType]:
     """Get detector data and apply user offsets to raw data.
 
     Parameters
     ----------
     dg:
         Data loaded with Mantid and converted to Scipp.
-    sample_offset:
-        Sample offset.
-    detector_bank_offset:
-        Detector bank offset.
     """
-    data = dg['data']
-    sample_pos = data.coords['sample_position']
-    sample_pos = sample_pos + sample_offset.to(unit=sample_pos.unit, copy=False)
-    pos = data.coords['position']
-    pos = pos + detector_bank_offset.to(unit=pos.unit, copy=False)
-    return RawDetector[RunType](
-        dg['data'].assign_coords(position=pos, sample_position=sample_pos)
-    )
+    # The generic NeXus workflow will try to extract 'data' from this, which is exactly
+    # what we also have in the Mantid data. We use the generic workflow since it also
+    # applies offsets, etc.
+    return NeXusDetector(dg)
 
 
-def assemble_detector_data(
+def get_monitor_data(
+    dg: LoadedFileContents[RunType], nexus_name: NeXusMonitorName[MonitorType]
+) -> NeXusMonitor[RunType, MonitorType]:
+    # The generic NeXus workflow will try to extract 'data' from this, which is exactly
+    # what we also have in the Mantid data. We use the generic workflow since it also
+    # applies offsets, etc.
+    monitor = dg['monitors'][nexus_name]['data']
+    return NeXusMonitor(sc.DataGroup(data=monitor, position=monitor.coords['position']))
+
+
+def dummy_assemble_detector_data(
     detector: CalibratedDetector[RunType],
 ) -> DetectorData[RunType]:
     """Dummy assembly of detector data, detector already contains neutron data."""
     return DetectorData[RunType](detector)
 
 
-def get_monitor_data(
-    dg: LoadedFileContents[RunType], nexus_name: NeXusMonitorName[MonitorType]
-) -> RawMonitor[RunType, MonitorType]:
-    # See https://github.com/scipp/sciline/issues/52 why copy needed
-    mon = dg['monitors'][nexus_name]['data'].copy()
```

Comment:
Is the need for this copy now gone? The issue is still open...

Reply:
I think we are not actually accessing a `value` property here?